src/main/java/com/goorm/jeff/global/interceptor/AuthInterceptor.java

```diff
@@ -0,0 +1,51 @@
+package com.goorm.jeff.global.interceptor;
+
+import com.goorm.jeff.global.exception.ForbiddenAuthException;
+import com.goorm.jeff.global.exception.NoAuthInfoException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpMethod;
+import org.springframework.web.servlet.HandlerInterceptor;
+import org.springframework.web.servlet.ModelAndView;
+
+
+
+@Slf4j
+public class AuthInterceptor implements HandlerInterceptor {
+
+    @Value("${key.api}")
+    private String apiKey;
```

Comment:
@Value 어노테이션은 이 자체로 null checking 이 안됩니다.
런타임에 프로그램이 죽을 수 있는 부분이기 때문에 외부 변수는 신경 써야 합니다.



```suggestion
@Configuration
@ConfigurationProperties(prefix = "key")
public class AppProperties {
    
    @NotNull
    @Valid // 중첩 객체 유효성 검사. JSR-380 Java Bean Validation 을 찾아보세요
    private Api api = new Api();

    public static class Api {
        @NotNull
        private String apiKey;
    }

    // 필요한 경우 추가적인 검증
    @PostConstruct
    public void validateProperties() {
        // 복잡한 검증 로직이 필요한 경우 @PostContruct 를 통해 구현 가능합니다.
    }
}
```


위와 같이 중첩 객체로 하지 않는 경우 더욱 손쉽게 구현 가능합니다.
@ConfigurationProperties 에 대해서 더 구글링해보세요